tests/console/yast2_lan_hostname_dhcp_yes_eth.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: yast2-network
+# Summary: Verify that correct value is stored in network config when
+# setting hostname via DHCP to 'yes: <ethernet network interface>' in
+# YaST2 lan module (https://bugzilla.suse.com/show_bug.cgi?id=984890)
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use parent 'yast2_lan_hostname_base';
+use strict;
+use warnings;
+use testapi;
+use scheduler qw(get_test_suite_data);
+use YaST::Module;
+
+sub run {
+    my $network_interface = script_output('ls /sys/class/net | grep ^e');
+    my $test_data         = get_test_suite_data();
+
+    YaST::Module::open({module => 'lan', ui => $test_data->{ui}});
```

Comment:
I can see that in order to not duplicate the test in the future (now only use for ncurses) you pass ui as parameter. I mentioned above that should be in some nested structure, for example related with lan, but also we would need to infer just reading the variable that 'ui' is the ui for a specific yast module to be open, not any other module. Could you represent that in test data? Another thought is that perhaps the function should not what to do, as that looks more like an environment variables, but I'm not sure, wdyt?

Reply:
Totally agree with nested structure, I'll apply it.

Regarding second question, I guess we cannot count this as environment variable, as, for example, this test is running on the system with Gnome installed, even though we test this module in ncurses.

When opening any yast2 module manually, a user can decide how to start it, with Qt or with ncurses by typing the appropriate commands in console. I also want to have this freedom while writing tests.